codeworld-api/src/CodeWorld/Color.hs

```diff
@@ -93,6 +97,30 @@ gray, grey :: Double -> Color
 gray = grey
 grey k = RGBA k k k 1
 
+
+-- Produce an infinite list of colors:
+-- https://github.com/google/codeworld/issues/436
+-- The list loops through the HSL spectrum with fixed
+-- saturation and lightness:
+-- https://www.w3schools.com/colors/colors_hsl.asp
+-- http://stackoverflow.com/a/7728626/1007926
+
+colorHues :: [Double]
+colorHues = let
+  start = 0.0
+  end = 2*pi
+  foldF :: (Double, Double) -> Maybe (Double, (Double, Double))
+  foldF (level, index) = let
+    pow2 = 2.0 ** (level - 1.0)
+    result = (end - start) * index / pow2
+    index' = mod' (index + pi/8) pow2
+    level' = if (index' == 1.0) then (level+1) else level
+    in Just (result, (level', index'))
+  in unfoldr foldF (1.0, 1.0)
+
+colors :: [Color]
```

Comment:
I'm a bit nervous about this name.  It's a good one, but there are 950 existing programs that define `colors` at module scope, and they will all break.  This is fewer than `rainbow`, at least; and `colors` is a less annoying name to steal.  But nevertheless, I wonder if there are better names that are less likely to break anything.

Ideas (none of which I like too well):

* `allColors`
* `assortedColors`
* `variousColors`
* `miscellaneousColors`